src/vtcomposite.cpp

```diff
@@ -692,104 +716,183 @@ struct LocalizeWorker : Napi::AsyncWorker
+                                }
+                                else
+                                {
+                                    skip_feature = true;
+                                }
                             }
 
-                            continue;
+                            // keep feature and retain its compatible worldview value
+                            else if (property_key == compatible_worldview_key)
+                            {
+                                has_worldview_key = true;
+
+                                if (property.value().type() == vtzero::property_value_type::string_value)
+                                {
+                                    std::string property_value = static_cast<std::string>(property.value().string_value());
+
+                                    std::vector<std::string> available_worldviews = utils::split(property_value);
+
+                                    // determine which worldviews to create a clone of the feature
+                                    if (keep_every_worldview)
+                                    {
+                                        worldviews_to_create = worldviews_for_feature(available_worldviews, available_worldviews);
+                                    }
+                                    else
+                                    {
+                                        worldviews_to_create = worldviews_for_feature(available_worldviews, baton_data_->worldviews);
+                                        if (worldviews_to_create.empty())
+                                        {
+                                            skip_feature = true;
+                                        }
+                                    }
+                                }
+                                else
+                                {
+                                    skip_feature = true;
+                                }
+                            }
+                            else // safeguard – should never reach here
+                            {
+                                skip_feature = true;
+                            }
                         }
 
-                        if (property_key == baton_data_->class_property)
+                        else if (
+                            utils::startswith(property_key, baton_data_->class_property) ||
+                            utils::startswith(property_key, baton_data_->class_prefix + baton_data_->class_property))
                         {
-                            if (property.value().type() == vtzero::property_value_type::string_value)
+                            // check if the property is of higher precedence that class key encountered so far
+                            std::uint32_t idx = static_cast<std::uint32_t>(std::find(class_key_precedence.begin(), class_key_precedence.end(), property_key) - class_key_precedence.begin());
```

Comment:
std::distance would be more idiomatic

Reply:
Changed to `std::distance` in 3711e05918f95ff3494302a0c36e07fe1bdfad08.